runtime/jcl/common/clsldr.cpp

```diff
@@ -116,9 +116,12 @@ Java_java_lang_ClassLoader_defineClassImpl1(JNIEnv *env, jobject receiver, jclas
 
 	jclass result = defineClassCommon(env, receiver, className, classRep, 0, length, protectionDomain, &options, hostClazz, NULL, FALSE);
 	if (env->ExceptionCheck()) {
+		throwNewInternalError(env, NULL);
 		return NULL;
 	} else if (NULL == result) {
-		throwNewInternalError(env, NULL);
+		vmFuncs->internalEnterVMFromJNI(currentThread);
+		vmFuncs->setCurrentException(currentThread, J9VMCONSTANTPOOL_JAVALANGNOCLASSDEFFOUNDERROR, (UDATA *)*(j9object_t *)className);
```

Comment:
How did we get out of defineClass without a pending exception?  I think we need to find that path and ensure it sets an exception as required by the contract for defineClassCommon - either a class is found or it returns null with pending exception

Reply:
It is fixed now. Directly return NULL when result is NULL.